pgcli/packages/sqlcompletion.py

```diff
@@ -30,7 +31,8 @@ def suggest_type(full_text, text_before_cursor):
     # partially typed string which renders the smart completion useless because
     # it will always return the list of keywords as completion.
     if word_before_cursor:
-        if word_before_cursor[-1] in ('(', '.'):
+        if (word_before_cursor[-1] in ('(', '.')
+                or word_before_cursor[0] == '\\'):
```

Comment:
There could be leading spaces that can thwart this logic. Try this for example: 

```
postgres>       \df <tab>
```

It doesn't trigger the right auto-completion. 

It might be worth stripping the leading white-space from word_before_cursor and then check for '\\'.